### ISSUE TITLE
Add Error highlighter functionality

### ISSUE THREAD

**Issue by agusibrahim** [akvelon/flutter-code-editor#139]
please error code hightlight or any sign to indicate an error. can be row or row and column.
thank.
![code error example](https://i.stack.imgur.com/txOB7.png)

**Comment by alexeyinkin** [comment]
## Option 1. Analyze the code in the frontend

This would require to run some sort of analyzer in Dart. The source of truth for errors is the native compiler / analyzer of that language. We could look for options to transpile compilers of some languages into Dart.

## Option 2. Supply a backend

The editor could send requests to a backend endpoint supplied by the app. We may supply wrappers for common compilers to expose them as endpoints, and add arguments to CodeController to specify endpoints. Or developers may wrap their own compilers.

**Comment by agusibrahim** [comment]
i've been used dart analyzer from dartpad
```
curl -X "POST" "https://stable.api.dartpad.dev/api/dartservices/v2/analyze" \
     -H 'Content-Type: text/plain; charset=utf-8' \
     -d "{\"source\":\"import 'package:flutter/materials.dart';\\n\\nvoid main() => runApp(const BottomSheetApp());\\n\\nclass BottomSheetApp extends StatelessWidget {\\n  const BottomSheetApp({super.key});\\n\\n  @override\\n  Widget build(BuildContext context) {\\n    return MaterialApp(\\n      home: Scaffold(\\n        appBar: AppBar(title: const Text('Bottom Sheet Sample')),\\n        body: const BottomSheetExample(),\\n      ),\\n    );\\n  }\\n}\\n\\nclass BottomSheetExample extends StatelessWidget {\\n  const BottomSheetExample({super.key});\\n\\n  @override\\n  Widget build(BuildContext context) {\\n    return Center(\\n      child: ElevatedButton(\\n        child: const Text('showModalBottomSheet'),\\n        onPressed: () {\\n          showModalBottomSheet<void>(\\n            context: context,\\n            builder: (BuildContext context) {\\n              return Container(\\n                height: 200,\\n                color: Colors.amber,\\n                child: Center(\\n                  child: Column(\\n                    mainAxisAlignment: MainAxisAlignment.center,\\n                    mainAxisSize: MainAxisSize.min,\\n                    children: <Widget>[\\n                      const Text('Modal BottomSheet'),\\n                      ElevatedButton(\\n                        child: const Text('Close BottomSheet'),\\n                        onPressed: () => Navigator.pop(context),\\n                      ),\\n                    ],\\n                  ),\\n                ),\\n              );\\n            },\\n          );\\n        },\\n      ),\\n    );\\n  }\\n}\\n\\n\\nvoid _result(bool success, [List<String> messages = const []]) {\\n  // Join messages into a comma-separated list for inclusion in the JSON array.\\n  final joinedMessages = messages.map((m) => '\\\"$m\\\"').join(',');\\n  print('__TESTRESULT__ {\\\"success\\\": $success, \\\"messages\\\": [$joinedMessages]}');\\n}\\n\\n// Ensure we have at least one use of `_result`.\\nvar resultFunction = _result;\\n\\n// Placeholder for unimplemented methods in dart-pad exercises.\\n// ignore: non_constant_identifier_names, sdk_version_never\\nNever TODO([String message = '']) => throw UnimplementedError(message);\\n\"}"
```

how to add red underline error in code? and pls add error message when clicking error icon (cross icon)
and, how to add text change handler?

**Comment by alexeyinkin** [comment]
Published in v0.2.10.
